monai/data/png_writer.py

```diff
@@ -62,8 +53,8 @@ def write_png(
             data = np.clip(data, _min, _max)
 
     if scale:
-        data = np.clip(data, 0.0, 1.0)  # png writer only can scale data in range [0, 1].
+        data = np.clip(data, 0.0, 1.0)  # png writer only can scale data in range [0, 1]
         data = 255 * data
-    data = data.astype(np.uint8)
-    io.imsave(file_name, data, plugin=plugin, **plugin_args)
+    img = Image.fromarray(data.astype(np.uint8))
```

Comment:
You can actually store unsigned 16bit alpha masks in PNG format, this is useful for storing MR images which are captured in that format typically. I wouldn't cast `data` to `uint8` here, I would leave up to the caller to pass a data array of a compatible format. PIL will complain if it's incompatible, so stating this requirement of `data` in the function documentation would be enough.

Reply:
Hi @ericspod ,

Thanks for your review. I see your point.
So let's only cast to `uint8` if scale 255?

Reply:
That is one way if only casting to uint8. Instead `scale` could be an optional numpy type value which when not None is used to rescale `data` by `np.iinfo(scale).max` after which `data` is cast to `scale`. MONAI is focused on biomedical applications so I'd suggest the default for `scale` should be uint16 to accommodate MR images as well as US and CT images shifted into an unsigned value range.

Reply:
Hi @ericspod ,

Thanks for your suggestions. I updated the code to support both `uint8` and `uint16` data type and keep the original type if no scaling.
I keep the original behavior to disable scaling by default, it's may be more straight-forward for users.
Could you please help review it again?

Reply:
Looks good me, though as a matter of good practice using something like `np.iinfo(np.uint8).max` in place of `255` in the code, or some named constant, avoids having magic numbers. Giving some name or identifying expression to a number rather than using a literal makes intent more explicit. Otherwise it's good to go.

Reply:
Good point, I updated the code to use `np.iinfo(np.uint8).max`.
Thanks.